lib/public/views/lhcPeriods/format/formatDistinctLhcBeamEnergies.js

```diff
@@ -13,81 +14,50 @@
-            : lhcBeamEnergy.toFixed(notChangedEnergyPrecision).toString();
+function getClosestEnergyLevel(lhcBeamEnergy) {
+    for (const energyLevel of WELL_DEFINED_ENERGY_LEVELS) {
+        if (lhcBeamEnergy < energyLevel - ENERGY_LEVEL_MARGIN) {
+            return null;
+        } else if (lhcBeamEnergy <= energyLevel + ENERGY_LEVEL_MARGIN) {
+            return energyLevel;
+        }
+    }
+    return null;
+}
 
 /**
- * Takes lhcBeamEnergy and return it or closest defined energy accoriding to the following conditions.
+ * Display a given LHC beam energy
  *
- *      If consider all energies `E DEFINED IN wellDefinedEnergyLevels`
- *      and take minimal absolute value of difference of lhcBeamEnergy and E
- *      and `|lhcBeamEnergy - closestEnergy| <= acceptableMargin`
- *      then the function returns `closestEnergy`
- *      if there is no `closestEnergy` fulfilling all requirements then the function returns `lhcBeamEnergy`.
+ * If the energy is close enough (within a given margin)
+ * to a well defined energy level, displays the well defined energy level instead
  *
- * @param {number} lhcBeamEnergy list of energies to be transformed
- * @returns {string} closest energy in displayable format
+ * @param {number} lhcBeamEnergy the energy to display
+ * @return {string} the energy display
  */
-const distinctLhcBeamEnergyToDisplay = (lhcBeamEnergy) => {
-    const closestEnergy = getClosestDefinedEnergyLevel(
-        lhcBeamEnergy,
-        Object.keys(distinctEnergiesProcessingConfiguration.wellDefinedEnergyLevelToDisplay),
-    );
-    return _distinctLhcBeamEnergyToDisplay(
-        lhcBeamEnergy,
-        closestEnergy,
-        distinctEnergiesProcessingConfiguration.wellDefinedEnergyLevelToDisplay,
-        distinctEnergiesProcessingConfiguration.acceptableEnergyMargin,
-    );
+const displayLhcBeamEnergy = (lhcBeamEnergy) => {
+    const energyLevel = getClosestEnergyLevel(lhcBeamEnergy);
+    const energyToDisplay = energyLevel !== null ? energyLevel : lhcBeamEnergy;
+    return energyToDisplay in ENERGY_LEVELS_CUSTOM_NAME
+        ? ENERGY_LEVELS_CUSTOM_NAME[energyToDisplay]
+        : energyToDisplay.toFixed(DISPLAY_PRECISION);
 };
 
 /**
- * Format lhc beam distinct energies, optionaly replacing some by closest energies
- * @param {number[]} distinctEnergies list of energies to be transformed
- * @returns {Component} list of distinct energies
+ * Format lhc beam distinct energies, optionaly replacing some of them
+ * by well defined energy levels @see displayLhcBeamEnergy
+ * @param {number[]} distinctEnergies list of energies to be displayed
+ * @returns {Component} formatted list of distinct energies
  */
 export const formatDistinctLhcBeamEnergies = (distinctEnergies) =>
     distinctEnergies?.length > 0
         ? h('', distinctEnergies
-            ?.map(distinctLhcBeamEnergyToDisplay)
+            ?.map(displayLhcBeamEnergy)
             .filter((value, index, array) => array.indexOf(value) === index) // Unique
```

Comment:
I don't know which one is the more optimized, but I would rather create a `Set` and go back to the array

Reply:
yes, sets are faster: https://medium.com/@jakubsynowiec/unique-array-values-in-javascript-7c932682766c

Reply:
This is what I thought